src/odb/src/db/dbDatabase.cpp

```diff
@@ -701,8 +702,8 @@ void dbDatabase::writeEco(dbBlock* block_, const char* filename)
 {
   _dbBlock* block = (_dbBlock*) block_;
 
-  FILE* file = fopen(filename, "w");
-  if (!file) {
+  std::ofstream file(filename, std::ios::binary);
```

Comment:
use exceptions, otherwise subsequent io operations won't be checked(such as running out of disk space).